src/Datadog.Trace.ClrProfiler.Native/clr_helpers.h

```diff
@@ -72,7 +74,7 @@ class EnumeratorIterator {
   inline T const& operator*() const { return arr_[idx_]; }
 
   inline EnumeratorIterator<T>& operator++() {
-    if (idx_ < sz_) {
+    if (idx_ < sz_ - 1) {
```

Comment:
This was a bug that caused us to look at one extra item in the array. We never went over the size of the array (256) so it never caused any issues, though.